src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAroundCheck.java

```diff
@@ -720,6 +739,32 @@ private boolean isEmptyCatch(DetailAST ast, int parentType) {
         return allowEmptyCatches && isEmptyBlock(ast, parentType, TokenTypes.LITERAL_CATCH);
     }
 
+    /**
+     * Tests if the given {@code DetailAst} is part of an allowed empty
+     * switch case or default block.
+     *
+     * @param ast the {@code DetailAst} to test.
+     * @return {@code true} if {@code ast} makes up part of an allowed
+     *         empty switch case or default block.
+     */
+    private boolean isEmptySwitchCaseOrDefault(DetailAST ast) {
+        final DetailAST parent = ast.getParent();
+        final DetailAST grandParent = parent.getParent();
+
+        final boolean isEmptyCaseInSwitchRule =
+                isEmptyBlock(ast, parent.getType(), TokenTypes.SWITCH_RULE);
+
+        final boolean isEmptyCaseGroupCheckedFromLcurly = isEmptyBlock(ast,
+                grandParent.getType(), TokenTypes.CASE_GROUP);
+
+        final boolean isEmptyCaseGroupCheckFromRcurly =
+                parent.getFirstChild().getType() == TokenTypes.RCURLY
+                  && grandParent.getParent().getType() == TokenTypes.CASE_GROUP;
+
+        return allowEmptySwitchCase && (isEmptyCaseInSwitchRule
```

Comment:
We should try to avoid all these other method calls if this property is false (i.e. delegate to another method or use predicates).